packages/react-instantsearch-hooks-dom/src/widgets/Highlight.tsx

```diff
@@ -0,0 +1,52 @@
+import {
+  getHighlightedParts,
+  getPropertyByPath,
+} from 'instantsearch.js/es/lib/utils';
+import React from 'react';
+
+import { Highlight as HighlightUiComponent } from '../ui';
+
+import type { HighlightProps as HighlightUiComponentProps } from '../ui';
+import type { BaseHit, Hit } from 'instantsearch.js';
+
+/**
+ * Make certain keys of an object optional
+ */
+type PartialKeys<TObj, TKeys extends keyof TObj> = Omit<TObj, TKeys> &
+  Partial<Pick<TObj, TKeys>>;
```

Comment:
We can probably already move it into a common `types/` directory to attract attention on it and avoid reimplementing it inadvertently in new PRs.